api_endpoints/api_auth/handler.py

```diff
@@ -0,0 +1,48 @@
+import os
+
+import psycopg2
+from psycopg2 import extras as pg_extras
+
+conn = psycopg2.connect(
+    host=os.environ.get("POSTGRES_HOST"),
+    database=os.environ.get("POSTGRES_DATABASE_NAME"),
+    user="postgres",
+    password=os.environ.get("POSTGRES_PASSWORD"),
+)
+cur = conn.cursor(cursor_factory=pg_extras.RealDictCursor)
+
+
+def lambda_handler(event, context):
+    if "token" not in event["queryStringParameters"]:
+        raise Exception("Unauthorized")
+    api_key = event["queryStringParameters"].get("token", None)
+    authentication = {"data": {"auth_token": api_key}}
+
+    cur.execute(
+        """
+        SELECT id, name, key
+        FROM users_apikey
+        WHERE key = %s
```

Comment:
Not sure there's a unique restraint on [key](https://github.com/DemocracyClub/ec-api-proxy/blob/65df7810c9ee076196c9b90a10ccf48a9f2de403/ec_api/apps/users/models.py#L69). Think there should be if doing things like this - can be punted to issue though.

Reply:
I think `fetchone()` will deal with this, but I've added a `LIMIT 1` to the query anyway :+1:

Reply:
I meant something slightly different. `fetchone` will mean you only get one `(id,name,key)` tuple returned, but I don't see what's stopping the same key being assigned to multiple users - so you might not get the _right_ tuple back. However I think this is unlikely to happen, hence can be an issue for later. Of course this might be handled somewhere that I haven't spotted/understood :)